### PR TITLE
[zstd] update to 1.5.0

### LINKED ISSUE #18912: [zstd] update to 1.5.0
Library name: zstd

New version number: 1.5.0

Other information that may be useful (release notes, etc...)

ref.:
https://github.com/facebook/zstd/releases

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #18912 

Update zstd to the latest version 1.5.0

Note: no feature need to test

### CODE REVIEW

**Comment by xavier2k6** [comment]
@JonLiu1993 are the regressions genuine?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 are the regressions genuine?

Sorry,I do not quite understand you.

**Comment by xavier2k6** [comment]
>Sorry, I do not quite understand you.

The test failures.

>##[error]REGRESSION: zstd:x64-uwp. If expected, add zstd:x64-uwp=fail to .\scripts\ci.baseline.txt.

>##[error]REGRESSION: zstd:arm-uwp. If expected, add zstd:arm-uwp=fail to .\scripts\ci.baseline.txt.

**Comment by JonLiu1993** [comment]
> > Sorry, I do not quite understand you.
> 
> The test failures.
> 
> > ##[error]REGRESSION: zstd:x64-uwp. If expected, add zstd:x64-uwp=fail to .\scripts\ci.baseline.txt.
> 
> > ##[error]REGRESSION: zstd:arm-uwp. If expected, add zstd:arm-uwp=fail to .\scripts\ci.baseline.txt.

I will fix it soon if I have time

File: ports/zstd/portfile.cmake [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-    SHA512 f529db9c094f9ae26428bf1fdfcc91c6d783d400980e0f0d802d2cf13c2be2931465ef568907e03841ff76a369a1447e7371f8799d8526edb9a513ba5c6db133
+    REF a488ba114ec17ea1054b9057c26a046fc122b3b6 #v1.5.0
+    SHA512 659576d0f52d2271b6b53f638b407b873888b1cffe4f014c3149d33a961653c2fcf7ff270bc669a5647205b573ef2809907645a4c89ab6c030ad65bce15547ae
     HEAD_REF dev
     PATCHES
       install_pkgpc.patch
+      fix-c4703-error.patch
 )
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```
Comment (NancyLi1013) [inline ports/zstd/portfile.cmake:12]:
```
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "static" ZSTD_STATIC )
string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" ZSTD_SHARED)
```
Please use the above way to choose static or dynamic build.

File: ports/zstd/portfile.cmake [hunk 2]
```diff
@@ -59,18 +59,16 @@ endif()
...
-        file(WRITE ${CURRENT_PACKAGES_DIR}/include/${HEADER} "${HEADER_CONTENTS}")
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/${HEADER}" "defined(ZSTD_DLL_IMPORT) && (ZSTD_DLL_IMPORT==1)" "1" )
     endforeach()
 endif()
 
-file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
-file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
+file(COPY "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
+file(COPY "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
 file(WRITE ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright "ZSTD is dual licensed - see LICENSE and COPYING files\n")
```
Comment (NancyLi1013) [inline ports/zstd/portfile.cmake:72]:
```suggestion
file(WRITE "${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright" "ZSTD is dual licensed - see LICENSE and COPYING files\n")
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by Tachi107** [comment]
@JonLiu1993 I believe that the `install_pkgpc` patch will become obsolete once/if https://github.com/facebook/zstd/pull/3252 gets merged. Notifying you just so you're aware.

**Pull request merged**